### PR TITLE
Amend indexes to accommodate msid and multiple versions

### PR DESCRIPTION
**Pull request by nlisgo**
This is a draft for now.

Just focusing on changes we can reasonably make to the manuscripts.txt and meca-lookup.txt so that we can prepare a data folder with the structure: `[msid]/v[version]`.

We need to allow for meca lookups for a second version of the manuscript.

### CODE REVIEW

File: manuscripts.txt [hunk 1]
```diff
@@ -1,79 +1,79 @@
...
+84792: 10.1101/2022.11.24.517802
+84831: 10.1101/2022.10.30.514301
+84855: 10.1101/2021.06.21.449261
+84874: 10.1101/2022.10.29.514266
+84875: 10.1101/2022.11.08.515699
+84918: 10.1101/2022.12.01.518662
+84921: 10.1101/2022.04.01.486717
+84926: 10.1101/2022.06.17.496451
+85069: 10.1101/2022.12.07.519455
+85111: 10.1101/2022.11.08.515698,10.1101/2022.11.08.515698
```
Comment (nlisgo) [inline manuscripts.txt:32]: If we have manuscript with multiple versions then repeat the preprint doi. If the preprint doi is different for the versions then just add the new preprint doi comma separated.

File: meca-lookup.txt [hunk 2]
```diff
@@ -5,4 +5,5 @@
 10.1101/2023.02.15.528649=s3://transfers-elife/biorxiv_Current_Content/February_2023/17_Feb_23_Batch_1526/4a8b2c2b-6e13-1014-8842-90e5ce4b1a0a.meca
 10.1101/2022.12.09.519720=s3://transfers-elife/biorxiv_Current_Content/February_2023/18_Feb_23_Batch_1527/f2ae30f0-6c00-1014-8600-9097ca768463.meca
 10.21203/rs.3.rs-2200020/v3=s3://prod-elife-epp-meca/2200020-meca.zip
-10.21203/rs.3.rs-2197876/v1=s3://prod-elife-epp-meca/2197876-meca.zip
\ No newline at end of file
+10.21203/rs.3.rs-2197876/v1=s3://prod-elife-epp-meca/2197876-meca.zip
+10.1101/2022.11.08.515698[2]=s3://transfers-elife/biorxiv_Current_Content/March_2023/21_Mar_23_Batch_1557/e7c056c2-6c16-1014-98e5-b5f7d8af8b03.meca
```
Comment (nlisgo) [inline meca-lookup.txt:9]: the square brackets with the version number in indicates to use this meca file rather than query bioRxiv api.

**Comment by nlisgo** [comment]
Wow. Infinity changes!!! :infinity:

**Comment by nlisgo** [comment]
The restructured data folder includes the `doi` and `[msid]/v[version]` structures so that we can serve both for the switch. Also, the files including the xml are the same as in the `doi` structure rather than the transforms being replayed. The xslt transform rules should still apply with the new structure because it grabs the doi from the xml rather than the folder structure.

**Comment by nlisgo** [comment]
The README.md has been updated to reflect the changes: https://github.com/elifesciences/enhanced-preprints-data/blob/msid-version-data-structure/README.md

**Comment by nlisgo** [comment]
I have run through a test of preparing the data folder from scratch before introducing these changes and preparing the new structure first and then running [./scripts/prepare_doi_data.sh](https://github.com/elifesciences/enhanced-preprints-data/blob/msid-version-data-structure/scripts/prepare_doi_data.sh) to prepare the data folder from the new structure and here is the diff:

```diff
Only in data-msid-version-doi/10.1101/2022.04.01.486717: 2022.04.01.486717.pdf
Only in data-msid-version-doi/10.1101/2023.02.08.527633: 2023.02.08.527633.pdf
Only in data-msid-version-doi/10.1101/2023.02.10.528013: 2023.02.10.528013.pdf
diff -r data-doi/10.21203/rs.3.rs-2197876/v1/source.txt data-msid-version-doi/10.21203/rs.3.rs-2197876/v1/source.txt
1c1
< 2197876-meca.zip
---
> 2197876-meca
diff -r data-doi/10.21203/rs.3.rs-2200020/v3/source.txt data-msid-version-doi/10.21203/rs.3.rs-2200020/v3/source.txt
1c1
< 2200020-meca.zip
---
> 2200020-meca
```

The source.txt differences are of no concern. I have checked and those 3 pdf's are in the data folder of the master branch but do not appear to be in the s3 bucket.

The 3 pdf's were introduced to the data repo here: https://github.com/elifesciences/enhanced-preprints-data/pull/68

The reason we don't have them in the prod s3 buckets is because the data folder has not been syncing since this PR has been merged. So, this would be expected. This means that my confidence in this PR being acceptable is high. I'm marking it as ready for review.

**Review by fred-atherden — APPROVED** [review]

**Pull request merged**